client/gui/CursorHandler.cpp

```diff
@@ -401,10 +405,16 @@ CursorHardware::~CursorHardware()
 
 void CursorHardware::setVisible(bool on)
 {
+#ifdef VCMI_APPLE
+	dispatch_async(dispatch_get_main_queue(), ^{

```

Comment:
Not an objection of any sort, but can you explain what is this and why it is needed?

Reply:
Quite a lot of UI frameworks prohibit to call UI from non-main thread

Reply:
Ah, this is wrapper that moves this call into main thread?
Sure, go ahead with commit, but I think this code can be moved to render() call which is always done in main thread.

Reply:
yes, SDL calls native API under the hood, and that one is supposed to be called from main thread (main "queue" is a synonym basically). And `^{ ... }` is objc "block" expression (= C++ lambda).